source/extensions/filters/network/rbac/rbac_filter.cc

```diff
@@ -63,9 +63,14 @@ RoleBasedAccessControlFilterConfig::RoleBasedAccessControlFilterConfig(
                                                   action_validation_visitor_)),
       shadow_engine_(Filters::Common::RBAC::createShadowEngine(
           proto_config, context, validation_visitor, action_validation_visitor_)),
-      enforcement_type_(proto_config.enforcement_type()) {}
+      enforcement_type_(proto_config.enforcement_type()),
+      delay_deny_ms_(PROTOBUF_GET_MS_OR_DEFAULT(proto_config, delay_deny, 0)) {}
 
 Network::FilterStatus RoleBasedAccessControlFilter::onData(Buffer::Instance&, bool) {
+  if (is_delay_denied_) {
+    return Network::FilterStatus::StopIteration;
+  }
```

Comment:
should we drain the data to avoid buffering the data when we wait the timer?

Reply:
Would it be good enough if we just disable the read after line 123 with `callbacks_->connection().readDisable(true);`? That way we stop reading any data if we determined the authz result is deny regardless if it is immediate or delay deny? cc @lambdai

Reply:
updated to disable the read.